### PR TITLE
MAGN-8016 Refactor mutation tester to use execution events

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

Spin waiting is replaced with execution events
[Task link](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8016)

### Reviewers

@pboyer, @lukechurch

### CODE REVIEW

File: src/DynamoCoreWpf/TestInfrastructure/AbstractMutator.cs [hunk 1]
```diff
@@ -11,13 +11,17 @@ abstract class AbstractMutator
     {
         //Convienence state, the presence of this state cache means that
         //usage of this mutator should be short lived
-        protected DynamoViewModel DynamoViewModel;
-        protected DynamoModel DynamoModel;
+        protected readonly DynamoViewModel DynamoViewModel;
+        protected readonly DynamoModel DynamoModel;
+
+        private const int MAX_TIME_WAIT = 20000;
```
Comment (lukechurch) [inline src/DynamoCoreWpf/TestInfrastructure/AbstractMutator.cs]: What unit is this in? ms I guess?

File: src/DynamoCoreWpf/TestInfrastructure/AbstractMutator.cs [hunk 2]
```diff
@@ -42,28 +45,25 @@ public virtual int NumberOfLaunches
...
 
-        // TODO(lukechurch): Move this to use event waiting rather than spin waiting
-        protected void SpinWaitForExecution()
+        private void WaitForExecution()
         {
-            while (!DynamoViewModel.HomeSpace.RunSettings.RunEnabled)
+            if (!executionWaitHandle.WaitOne(MAX_TIME_WAIT))
             {
-                Thread.Sleep(10);
+                throw new TimeoutException("Execution has been taking too long");
```
Comment (lukechurch) [inline src/DynamoCoreWpf/TestInfrastructure/AbstractMutator.cs]: s/Execution has been taking too long/Mutation Test timed out: + MAX_TIME_WAIT + ms

**Comment by lukechurch** [comment]
lgtm with nits. @pboyer

**Comment by marimano** [comment]
Hi @pboyer , could you please take a look?

**Comment by Racel** [comment]
@pboyer @lukechurch - Can you please take a look at this? @marimano and I discussed it this morning and were wondering why it did not get merged.

**Comment by pboyer** [comment]
@lukechurch This one looks good to me. Thoughts?

**Comment by pboyer** [comment]
Ah, @lukechurch I see your comment now. Merging...

**Pull request merged**